e2e/util/config.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+)
+
+type TestConfig struct {
+	ChannelName      string
+	ChannelNamespace string
+	GitURL           string
+}
+
+var config = &TestConfig{}
+
+func ReadConfig() error {
+	viper.SetDefault("ChannelName", defaultChannelName)
+	viper.SetDefault("ChannelNamespace", defaultChannelNamespace)
+	viper.SetDefault("GitURL", defaultGitURL)
+
+	if err := viper.BindEnv("ChannelName", "ChannelName"); err != nil {
+		return (err)
+	}
+
+	if err := viper.BindEnv("ChannelNamespace", "ChannelNamespace"); err != nil {
+		return (err)
+	}
+
+	if err := viper.BindEnv("GitURL", "GitURL"); err != nil {
+		return (err)
+	}
+
```

Comment:
e2e from GH action overwrites this config.yaml [here](https://github.com/RamenDR/ramen/blob/797c1a81a5cb653db40bffb15cbbecd3ce3532bc/.github/workflows/e2e.yaml#L68) is the YAML format the same, such that this works?

Reply:
Yes, the yaml format is the same. if we have a config.yaml like below, I could change code and handle it too.
```
channelname: "ramen-gitops"
channelnamespace: "ramen-samples"
giturl: "https://github.com/RamenDR/ocm-ramen-samples.git"
Clusters:
  c1:
    kubeconfigpath: /home/par/.config/drenv/rdr-rdr/kubeconfigs/rdr-dr1
  c2:
    kubeconfigpath: /home/par/.config/drenv/rdr-rdr/kubeconfigs/rdr-dr2
  hub:
    kubeconfigpath: /home/par/.config/drenv/rdr-rdr/kubeconfigs/rdr-hub
```

Reply:
right now the GH action overwrites this config.yaml, with only the cluster kubeconfig info, so my code will use default settings for channelname, channelnamespace, giturl.

Reply:
later we could update the git action, instead of directly copy, we could let it just add the kubeconfig file to the existing config.yaml, like
`cat ~/.config/drenv/${{ env.NAME_PREFIX }}rdr/config.yaml >> e2e/config.yaml`

Reply:
Understood. This works as the defaults are in the code at present.